libs/util/network.hpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_NETWORK_HPP_
+#define IROHA_NETWORK_HPP_
+
+#include <cstdint>
+#include <string>
+#include "string.hpp"
+
+namespace iroha {
+  namespace network {
+
+    /**
+     * @tparam T integer type
+     * @return true if 2^16 > port > 0, false otherwise.
+     */
+    template<typename T>
+    constexpr inline bool is_port_valid(T port) {
```

Comment:
why templated? any number that contains port can be casted to int safely

Reply:
The problem will occur when user wants to use is_port_valid for big ints, like `uint64`:
```
// Example program
#include <iostream>
#include <string>
#include <type_traits>
#include <limits>
#include <cassert>

template<typename T>
constexpr inline bool is_port_valid(T port) {
  return std::numeric_limits<T>::is_integer && port > 0 && port < 65536;
}

int main()
{
  uint64_t a = 8589940024;
  uint16_t b = a;
  
  std::cout << b; // 5432
  
  assert(is_port_valid(a) == false); // assert ok
}

```

Reply:
Probably I will change to just `uint64_t port`